src/openrct2/network/network.cpp

```diff
@@ -1855,27 +1858,42 @@ void Network::Server_Handle_GAMECMD(NetworkConnection& connection, NetworkPacket
 
 	sint32 commandCommand = args[4];
 
-	sint32 ticks = SDL_GetTicks(); //tick count is different by time last_action_time is set, keep same value.
+	uint32 ticks = SDL_GetTicks(); //tick count is different by time last_action_time is set, keep same value.
 
 	// Check if player's group permission allows command to run
 	NetworkGroup* group = GetGroupByID(connection.Player->Group);
 	if (!group || (group && !group->CanPerformCommand(commandCommand))) {
 		Server_Send_SHOWERROR(connection, STR_CANT_DO_THIS, STR_PERMISSION_DENIED);
 		return;
 	}
+
 	// In case someone modifies the code / memory to enable cluster build,
 	// require a small delay in between placing scenery to provide some security, as
 	// cluster mode is a for loop that runs the place_scenery code multiple times.
 	if (commandCommand == GAME_COMMAND_PLACE_SCENERY) {
-		if ((ticks - connection.Player->LastActionTime) < 20) {
+		if (
+			ticks - connection.Player->LastPlaceSceneryTime < ACTION_COOLDOWN_TIME_PLACE_SCENERY &&
+			ticks > connection.Player->LastPlaceSceneryTime
+		) {
 			if (!(group->CanPerformCommand(MISC_COMMAND_TOGGLE_SCENERY_CLUSTER))) {
-				Server_Send_SHOWERROR(connection, STR_CANT_DO_THIS, STR_CANT_DO_THIS);
+				Server_Send_SHOWERROR(connection, STR_CANT_DO_THIS, STR_NETWORK_ACTION_RATE_LIMIT_MESSAGE);
 				return;
 			}
 		}
 	}
+	// This is to prevent abuse of demolishing rides. Anyone that is not the server
+	// host will have to wait a small amount of time in between deleting rides.
+	else if (commandCommand == GAME_COMMAND_DEMOLISH_RIDE) {
+		if (
+			ticks - connection.Player->LastDemolishRideTime < ACTION_COOLDOWN_TIME_DEMOLISH_RIDE &&
+			ticks > connection.Player->LastDemolishRideTime
```

Comment:
Also I've been thinking of something.. Instead of having "LastDemolishRideTime" and "LastPlaceSceneryTime", what if there was just an array called LastCommandTimes? That way it could only need one if check, rather than checking for specific commands first and then checking the cooldown times? and add room for future cooldown times (maybe even a custom timeout interface for server admins at some point)? Just an idea.. I'm not sure if it would be worth all the array indices at this point though, so I'm asking you first for your opinion.